test/framework/functional/base_functional_test_case.py

```diff
@@ -71,9 +71,20 @@ def tearDown(self):
 
         # Kill processes and make sure all processes exited with 0 exit code
         services = self.cluster.kill()
-        for service in services:
-            self.assertEqual(service.return_code, 0, 'Service running on url: {} should exit with code 0, but exited '
-                                                     'with code {}.'.format(service.url, service.return_code))
+
+        # only check the exit code if not on Windows as Popen.terminate kills the process on Windows and the exit
+        # code is not zero.
+        # TODO: remove the os.name check after we can handle exit on Windows gracefully.
+        if os.name != 'nt':
```

Comment:
I know this is theoretically temporary, but want to change this to the nice is_windows helper you added? :wink: